src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -53,6 +52,7 @@ private sealed class State
 
             public bool Connected;
             public long AbortErrorCode = -1;
+            public int StreamCount;
```

Comment:
Can we instead have the stream safe handle reference the connection safe handle, using `DangerousAddReference` etc.?

Reply:
I was thinking about this more since our conversation. I was hesitant to embark on the Dangerous* route. While it may work for SafeHandle itself I feel that would be problematic for the GCHandle as it does not have buit-in ref count. And we may need the link for other reasons.